### ISSUE TITLE
ENABLED_PLUGINS is misleading. It should be an entry_point whitelist instead.

### ISSUE THREAD

**Issue by nrb** [shaunduncan/helga#136]
I have the following plugins enabled:

```
ENABLED_PLUGINS = [
        'manager',
        'help',
        'meant_to_say',
        'operator',
        'facts',
        'oneliner',
            'nightfall',
        'heroic',
        'daily',
    ]
```

Here's what I see in the plugin list:

```
palendae: !plugins list
dinklebot: Plugins enabled on this channel: daily, facts, giphy, help, heroic, manager, meant_to_say, nightfall, oneliner, operator
dinklebot: Available plugins: dubstep, icanhazascii, jira, loljava, no_more_olga, poems, reminders, reviewboard, stfu, webhooks, wiki_whois
```

Notice that poems, dubstep, icanhazascii, loljava, jira, reminders, reviewboard, stfu, webhooks, and wiki_whois are enabled.

Trying to `!enable poems`, for example, doesn't return an acknowledgement, and the `!haiku` commands don't return any output. This includes `!haiku add_use fives/sevens`/ `!haiku`. I've added lines, but `!haiku` still doesn't return output. I have not yet looked in the DB manually.

My expectation is that when a core/default plugin is not enabled, it's not available.

**Comment by shaunduncan** [comment]
Let me take a look at this. I think there might be a couple of things. ENABLED_PLUGINS is a bit of a misnomer. The plugin loader will grab any plugin defined in an entry point, but this setting is more about setting what plugins should be enabled by default for a channel. So if it's not in that list, it will be 'available' to be enabled via `!plugins enable foo`

**Comment by nrb** [comment]
Some more example use from my channel - this is all uninterrupted.

```
palendae: !haiku
palendae: !enable poems
palendae: !haiku
palendae: !haiku add sevens She is really hard to kill
palendae: !haiku add fives Omnigul the Witch
palendae: !haiku add fives But we will prevail
palendae: !haiku

**Comment by shaunduncan** [comment]
Can you try `!plugins enable poems` instead and see if that works?

**Comment by nrb** [comment]
That worked.

**Comment by shaunduncan** [comment]
Cool, I think I'm going to chock this up as an enhancement. ENABLED_PLUGINS is definitely misleading, so there should be two distinct settings:

- ENABLED_PLUGINS: A list of entry_point names that would whitelist plugins that should be loaded from an entry_point and made available for use. None would imply they all are made available.
- DEFAULT_CHANNEL_PLUGINS: This would do what ENABLED_PLUGINS does now, define the plugin names that are enabled by default on a channel.

I'll probably want to make this for webhook settings as well (or at least ensure that it works the same way).

**Comment by shaunduncan** [comment]
Fixed in the 1.7.0 release. There are 3 settings:

- ENABLED_PLUGINS: serves as a plugin whitelist
- DISABLED_PLUGISN: serves as a plugin blacklist
- DEFAULT_CHANNEL_PLUGINS: serves as a list of plugins that should be 'on' by default in a channel
